lib/bap/bap_synthetic_symbolizer.ml

```diff
@@ -0,0 +1,142 @@
+open Core_kernel.Std
+open Bap_types.Std
+open Bap_image_std
+open Bap_disasm_std
+open Bap_sema.Std
+open Monads.Std
+
+include Bap_self.Create()
+
+module Fact = Ogre.Make(Monad.Ident)
+
+module Ext = struct
+  open Image.Scheme
+  open Fact.Syntax
+
+  let addr_width =
+    Fact.require arch >>= fun a ->
+    match Arch.of_string a with
+    | Some a -> Fact.return (Arch.addr_size a |> Size.in_bits)
+    | None -> Fact.failf "unknown/unsupported architecture" ()
+
+  let calls insns =
+    let insns =
```

Comment:
bad variable naming: the are not instructions, but instructions that may affect control flow. Though it is also not right.